website/docs/r/cdn_frontdoor_secret.html.markdown

```diff
@@ -0,0 +1,143 @@
+---
+subcategory: "CDN"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_cdn_frontdoor_secret"
+description: |-
+  Manages a Frontdoor Secret.
+---
+
+# azurerm_cdn_frontdoor_secret
+
+Manages a Frontdoor Secret.
+
+## Required Key Vault Permissions
+
+!>**IMPORTANT:** You must add an `Access Policy` to your `azurerm_key_vault` for the `Microsoft.AzureFrontDoor-Cdn` Enterprise Application Object ID.
+
+| Object ID                                | Key Permissions | Secret Permissions   | Certificate Permissions                       |
+|:-----------------------------------------|:---------------:|:--------------------:|:---------------------------------------------:|
+| `Microsoft.AzureFrontDoor-Cdn` Object ID | -               | **Get**              | -                                             |
+| Your Personal AAD Object ID              | -               | **Get** and **List** | **Get**, **List**, **Purge** and **Recover**  |
+| Terraform Service Principal              | -               | **Get**              | **Get**, **Import**, **Delete** and **Purge** |
+
+->**NOTE:** You only need to add the `Access Policy` for your personal AAD Object ID if you are planning to view the `secrets` via the Azure Portal.
+
+## Example Usage
+
+```hcl
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_key_vault" "example" {
+  name                       = "example-keyvault"
+  location                   = azurerm_resource_group.example.location
+  resource_group_name        = azurerm_resource_group.example.name
+  tenant_id                  = data.azurerm_client_config.current.tenant_id
+  sku_name                   = "premium"
+  soft_delete_retention_days = 7
+
+  network_acls {
+    default_action = "Deny"
+    bypass         = "AzureServices"
+    ip_rules       = ["10.0.0.0/24"]
+  }
+
+  # Frontdoor Enterprise Application Object ID(e.g. Microsoft.AzureFrontDoor-Cdn)
+  access_policy {
+    tenant_id = data.azurerm_client_config.current.tenant_id
+    object_id = "00000000-0000-0000-0000-000000000000" # <- Object Id for the Microsoft.AzureFrontDoor-Cdn Enterprise Application
```

Comment:
Can we show how to programatically get this in the example via a data source?

Reply:
I am not exactly sure how to do that since I do not have access to AzureAD in my tenant... 🙁